### PR TITLE
Document CannotFindDerivingType Error

### PR DESCRIPTION
**Pull request by therewillbecode**
Add description of error as well as an example of a common scenario where the error can occur.

### CODE REVIEW

File: errors/CannotFindDerivingType.md [hunk 1]
```diff
@@ -5,16 +5,39 @@
 ```purescript
 module ShortFailingExample where
 
+type Entry =
+  { firstName :: String
+  , lastName  :: String
+  , address   :: Address
```
[the remark this answers is not in the document]
Comment (therewillbecode) [reply errors/CannotFindDerivingType.md]: That's right the Address type isn't defined in the example. I'll remove it.

File: errors/CannotFindDerivingType.md [hunk 2]
```diff
@@ -5,16 +5,39 @@
...
+  }
+
+derive instance eqEntry :: Eq Entry
 ...
 ```
 
 ## Cause
 
-Explain why a user might see this error.
+The type declaration for the given type is not in scope.
```
[the remark this answers is not in the document]
Comment (therewillbecode) [reply errors/CannotFindDerivingType.md]: That is a far more informative message. Much better.

[the remark this answers is not in the document]
Comment (chexxor) [reply errors/CannotFindDerivingType.md]: @therewillbecode Should I wait for you to update that part, also?

[the remark this answers is not in the document]
Comment (therewillbecode) [reply errors/CannotFindDerivingType.md]: Sure I'll do that.

**Comment by hdgarrood** [comment]
I'm going to close this, because it's a little stale, but also because I think this is actually a minor compiler bug; in this case, we should be throwing a `TypeSynonymInstance` error ("Type class instances for type synonyms are disallowed.") as opposed to this one.

**Pull request closed without merging**